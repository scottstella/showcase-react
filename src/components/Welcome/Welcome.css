.welcome-logo {
  height: 108px;
  width: 192px;
  align-self: center;
}

.welcome h2 {
  font-size: 3.5em;
  max-width: 800px;
}
.welcome p {
  margin-top: 30px;
  line-height: 2em;
  color: #777;
}

.welcome-header-section {
  display: flex;
  width: 100%;
}
