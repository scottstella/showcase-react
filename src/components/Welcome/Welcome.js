import React from "react";
import "./Welcome.css";
import hearthstoneIcon from "../../assets/images/hearthstone-icon.png";

export default function Welcome() {
  return (
    <div className="welcome">

     <div className="welcome-header-section">
      <img className="welcome-logo" src={hearthstoneIcon} alt="Hearthstone Icon"/>
      <h2>Welcome</h2>
      </div> 
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Et voluptas
        eligendi maiores beatae eius, odit aperiam, pariatur tenetur sequi quis
        doloribus perspiciatis neque voluptatem, consequuntur aliquid quam.
        Voluptatem perferendis ea reiciendis impedit similique cupiditate
        explicabo necessitatibus! Minima velit, praesentium rerum sed delectus
        excepturi, ducimus ipsum repudiandae suscipit ullam at consequuntur
        aliquid fuga iure dicta nisi ex sequi. Porro ipsum laborum ut velit
        nulla fugit officiis omnis, eos repudiandae saepe veniam molestiae. Enim
        officia aut nemo voluptatum ratione sapiente ipsa obcaecati cum, quas
        veritatis debitis quis odit quasi minus explicabo incidunt!
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Et voluptas
        eligendi maiores beatae eius, odit aperiam, pariatur tenetur sequi quis
        doloribus perspiciatis neque voluptatem, consequuntur aliquid quam.
        Voluptatem perferendis ea reiciendis impedit similique cupiditate
        explicabo necessitatibus! Minima velit, praesentium rerum sed delectus
        excepturi, ducimus ipsum repudiandae suscipit ullam at consequuntur
        aliquid fuga iure dicta nisi ex sequi. Porro ipsum laborum ut velit
        nulla fugit officiis omnis, eos repudiandae saepe veniam molestiae. Enim
        officia aut nemo voluptatum ratione sapiente ipsa obcaecati cum, quas
        veritatis debitis quis odit quasi minus explicabo incidunt!
      </p>
      <p>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus
        repudiandae reiciendis accusamus maxime dolor eum in sed quam hic sunt
        ad veniam architecto harum, dolorum eveniet distinctio provident nam at
        placeat asperiores! Eum veniam molestias ab odit, necessitatibus ipsa
        quam veritatis perferendis doloribus dicta voluptatibus voluptatum
        excepturi ullam quasi sed maxime quaerat qui consequuntur earum illo
        labore officiis nemo impedit facere. Veritatis numquam excepturi dolor
        vero rem. Voluptates eligendi ut voluptate. Ullam magni quisquam
        voluptatibus veniam autem? Sed iure ipsam odio vel porro distinctio
        vitae eum eos possimus ab, doloribus perspiciatis omnis ut impedit fugit
        earum voluptatem eligendi quidem voluptatum?
      </p>
    </div>
  );
}
