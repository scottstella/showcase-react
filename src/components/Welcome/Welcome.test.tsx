import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Welcome from "./Welcome";

describe("Welcome", () => {
  it("renders without crashing", () => {
    render(<Welcome />);
    expect(screen.getByText("Welcome")).toBeInTheDocument();
  });

  it("displays the Hearthstone icon with correct attributes", () => {
    render(<Welcome />);
    const logo = screen.getByAltText("Hearthstone Icon");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute(
      "src",
      "/src/assets/images/hearthstone-icon.png"
    );
    expect(logo).toHaveClass("welcome-logo");
  });

  it("has correct structure and styling", () => {
    render(<Welcome />);

    // Check main container
    const container = screen.getByTestId("welcome-container");
    expect(container).toHaveClass("welcome");

    // Check header section
    const headerSection = screen.getByTestId("welcome-header");
    expect(headerSection).toHaveClass("welcome-header-section");

    // Check heading
    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toHaveTextContent("Welcome");

    // Check paragraphs (using container query since paragraphs don't have a role)
    const paragraphs = container.querySelectorAll("p");
    expect(paragraphs).toHaveLength(3);
    paragraphs.forEach((p) => {
      expect(p).toBeInTheDocument();
      expect(p).toHaveTextContent(/Lorem ipsum/);
    });
  });

  it("matches snapshot", () => {
    const { container } = render(<Welcome />);
    expect(container).toMatchSnapshot();
  });
});
