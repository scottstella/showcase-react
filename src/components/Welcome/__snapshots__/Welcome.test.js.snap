// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Welcome > matches snapshot 1`] = `
<div>
  <div
    class="welcome"
    data-testid="welcome-container"
  >
    <div
      class="welcome-header-section"
      data-testid="welcome-header"
    >
      <img
        alt="Hearthstone Icon"
        class="welcome-logo"
        src="/src/assets/images/hearthstone-icon.png"
      />
      <h2>
        Welcome
      </h2>
    </div>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Et voluptas eligendi maiores beatae eius, odit aperiam, pariatur tenetur sequi quis doloribus perspiciatis neque voluptatem, consequuntur aliquid quam. Voluptatem perferendis ea reiciendis impedit similique cupiditate explicabo necessitatibus! Minima velit, praesentium rerum sed delectus excepturi, ducimus ipsum repudiandae suscipit ullam at consequuntur aliquid fuga iure dicta nisi ex sequi. Porro ipsum laborum ut velit nulla fugit officiis omnis, eos repudiandae saepe veniam molestiae. Enim officia aut nemo voluptatum ratione sapiente ipsa obcaecati cum, quas veritatis debitis quis odit quasi minus explicabo incidunt.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Et voluptas eligendi maiores beatae eius, odit aperiam, pariatur tenetur sequi quis doloribus perspiciatis neque voluptatem, consequuntur aliquid quam. Voluptatem perferendis ea reiciendis impedit similique cupiditate explicabo necessitatibus! Minima velit, praesentium rerum sed delectus excepturi, ducimus ipsum repudiandae suscipit ullam at consequuntur aliquid fuga iure dicta nisi ex sequi. Porro ipsum laborum ut velit nulla fugit officiis omnis, eos repudiandae saepe veniam molestiae. Enim officia aut nemo voluptatum ratione sapiente ipsa obcaecati cum, quas veritatis debitis quis odit quasi minus explicabo incidunt!
    </p>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Minus repudiandae reiciendis accusamus maxime dolor eum in sed quam hic sunt ad veniam architecto harum, dolorum eveniet distinctio provident nam at placeat asperiores! Eum veniam molestias ab odit, necessitatibus ipsa quam veritatis perferendis doloribus dicta voluptatibus voluptatum excepturi ullam quasi sed maxime quaerat qui consequuntur earum illo labore officiis nemo impedit facere. Veritatis numquam excepturi dolor vero rem. Voluptates eligendi ut voluptate. Ullam magni quisquam voluptatibus veniam autem? Sed iure ipsam odio vel porro distinctio vitae eum eos possimus ab, doloribus perspiciatis omnis ut impedit fugit earum voluptatem eligendi quidem voluptatum?
    </p>
  </div>
</div>
`;
